pyro/poutine/mask_messenger.py

```diff
@@ -31,5 +32,9 @@ def __init__(self, mask):
         self.mask = mask
 
     def _process_message(self, msg):
+        if msg["done"] or msg["type"] != "sample":
+            return None
+
         msg["mask"] = self.mask if msg["mask"] is None else self.mask & msg["mask"]
+        msg["fn"] = MaskedDistribution(msg["fn"], msg["mask"])
```

Comment:
If we replaced this with `msg["fn"].mask(msg["mask"])` then we could add short-circuit logic to avoid multiply wrapping a distribution in `MaskedDistribution. Just a thought.